src/main/java/org/candlepin/async/tasks/CloudAccountOrgSetupJob.java

```diff
@@ -0,0 +1,227 @@
+        }
+    }
+
+    /**
+     * Creates a JobConfig configured to execute the isAnonymous register job. Callers may further
+     * manipulate the JobConfig as necessary before queuing it.
+     *
+     * @return a JobConfig instance configured to execute the isAnonymous register job
+     */
+    public static CloudAccountOrgSetupJobConfig createJobConfig() {
+        return new CloudAccountOrgSetupJobConfig();
+    }
+
+    /**
+     * Job configuration object for the export job
+     */
+    public static class CloudAccountOrgSetupJobConfig extends JobConfig<CloudAccountOrgSetupJobConfig> {
+
+        public CloudAccountOrgSetupJobConfig() {
+            this.setJobKey(JOB_KEY).setJobName(JOB_NAME).addConstraint(
+                JobConstraints.uniqueByArguments(CLOUD_ACCOUNT_ID, OFFERING_ID)).setRetryCount(5);
+        }
+
+        /**
+         * Sets the cloud account ID for this job
+         *
+         * @param cloudAccountId
+         *     the cloud account ID to set for this job
+         * @return a reference to this job config
+         */
+        public CloudAccountOrgSetupJobConfig setCloudAccountId(String cloudAccountId) {
+            if (cloudAccountId == null || cloudAccountId.isEmpty()) {
+                throw new IllegalArgumentException("cloudAccountId is null or empty");
+            }
+
+            this.setJobArgument(CLOUD_ACCOUNT_ID, cloudAccountId);
+
+            return this;
+        }
+
+        /**
+         * Sets the cloud offering ID for this job
+         *
+         * @param cloudOfferingId
+         *     the cloud offering ID to set for this job
+         * @return a reference to this job config
+         */
+        public CloudAccountOrgSetupJobConfig setCloudOfferingId(String cloudOfferingId) {
+            if (cloudOfferingId == null || cloudOfferingId.isEmpty()) {
+                throw new IllegalArgumentException("cloudOfferingId is null or empty");
+            }
+
+            this.setJobArgument(OFFERING_ID, cloudOfferingId);
+
+            return this;
+        }
+
+        /**
+         * Sets the cloud provider for this job
+         *
+         * @param cloudProviderShortName
+         *     shortcut of the cloud provider
+         * @return a reference to this job config
+         */
+        public CloudAccountOrgSetupJobConfig setCloudProvider(CloudProvider cloudProviderShortName) {
+            if (cloudProviderShortName == null) {
+                throw new IllegalArgumentException("cloudProviderShortName is null");
+            }
+
+            this.setJobArgument(CLOUD_PROVIDER, cloudProviderShortName);
+
+            return this;
+        }
+
+        /**
+         * Sets the owner key for this job
+         *
+         * @param ownerKey
+         *     the owner key to set for this job
+         * @return a reference to this job config
+         */
+        public CloudAccountOrgSetupJobConfig setOwnerKey(String ownerKey) {
+            if (ownerKey.isEmpty()) {
```

Comment:
This will throw a NPE if the ownerKey is null, so let's make sure that `ownerKey != null` first in this if statement

Reply:
done

Reply:
Instead of 2 nested ifs, can we do it in 1? it is cleaner this way: `if(ownerKey !=null && ownerKey.isBlank())`

Reply:
done